src/modules/Library/features/Popups/AddToBuilderPopup/AddToBuilderPopup.tsx

```diff
@@ -73,6 +67,12 @@ export const AddToBuilderPopup = ({
     resolver: yupResolver(validationSchema[step]),
   });
 
+  const navigateToBuilder = (appletId: string, data: SingleApplet) => {
+    navigate(generatePath(page.builderAppletAbout, { appletId }), {
+      state: { isFromLibrary: true, data },
```

Comment:
could you confirn if you navigate from a library (with template) then navigate to dashboard and open an applet to edit, will it get proper values and will  `state` entirely (or `isFromLibrary` and `data`) be removed?

Reply:
any action like clicking a Link or throught navigate will clear location.state